### PR TITLE
mypy: warn_unused_ignores = true (and remove some)

### PR DESCRIPTION
**Pull request by altendky**
https://mypy.readthedocs.io/en/stable/config_file.html#confval-warn_unused_ignores

This PR should not change any code, nor any hints.  It should

1. remove unused type ignore comments
2. enforce checking for unused type ignores in CI
3. install our `dev` extra dependencies in CI for checks
4. add a few `dev` extra dependencies
5. deal with pre-commit's limitations for now by doing a looser check
   - See https://github.com/Chia-Network/chia-blockchain/pull/8882 if you want it more strict in pre-commit

### CODE REVIEW

**Review by almogdepaz — DISMISSED** [review]
lgtm

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -22,3 +22,7 @@ repos:
     hooks:
     - id: mypy
       additional_dependencies: [types-setuptools, types-PyYAML]
+      # This intentionally counters the settings in mypy.ini to allow a loose local
+      # check and a strict CI check.  This difference may or may not be retained long
```
Comment (xdustinface) [inline .pre-commit-config.yaml:26]:
```suggestion
      # check and a strict CI check. This difference may or may not be retained long
```

nit, so you may ignore it :)

**Review by xdustinface — DISMISSED** [review]
:+1:

**Review by xdustinface — DISMISSED** [review]

**Comment by altendky** [comment]
Kick CI for a fresh merge and build.

**Review by xdustinface — APPROVED** [review]

**Pull request merged**